src/box/memtx_tx.c

```diff
@@ -1769,6 +1795,91 @@ memtx_tx_handle_point_hole_write(struct space *space, struct memtx_story *story,
 	mh_point_holes_del(ht, pos, 0);
 }
 
+static bool
+memtx_tx_tuple_matches(struct index *index, struct tuple *tuple,
+		       enum iterator_type type, const char *key,
+		       uint32_t part_count)
+{
+	if (key == NULL) {
+		assert(part_count == 0);
+		assert(type == ITER_LE || type == ITER_GE);
+
+		/* An empty key matches to any tuple. */
+		return true;
+	}
+
+	struct key_def *def = index->def->key_def;
+	hint_t key_hint = def->key_hint(key, part_count, def);
+	hint_t tuple_hint = def->tuple_hint(tuple, def);
+	int cmp = def->tuple_compare_with_key(tuple, tuple_hint, key,
+					      part_count, key_hint, def);
```

Comment:
Nit: Please use wrappers instead of calling `key_def` members directly.

Reply:
Done.